providers/resource/bin/graphics_card_resource.py

```diff
@@ -170,7 +189,12 @@ def main():
                 record['driver'] = 'unknown'
             # lp:1636060 – If discrete GPU is using amdgpu driver,
             # we set the prime_gpu_offload flag to 'On'
-            if index == 2 and record['driver'] == 'amdgpu':
+            # NVIDIA driver since version 435.17 supports PRIME render offload,
+            # and Ubuntu doesn't support intel mode after 22.04.
+            if index == 2 and (record['driver'] == 'amdgpu'
+                               or (get_ubuntu_version() >= 22.04
+                                   and (record['driver'] == 'nvidia'
+                                        or record['driver'] == 'pcieport'))):
```

Comment:
The last part could be simplified with:

```
record['driver'] in ('nvidia', 'pcieport')
```

What do you think?

Reply:
This is great. I'm going to fix it.